crates/polars-lazy/src/physical_plan/executors/scan/mod.rs

```diff
@@ -84,6 +84,9 @@ impl Executor for DataFrameExec {
                 state.insert_has_window_function_flag()
             }
             let s = selection.evaluate(&df, state)?;
+            if state.cache_window() {
```

Comment:
See comments in AnonymousScan

Reply:
It seems that `DataStreamScan` already has `predicate_has_windows` field.

Reply:
Oh great. Then we can use that. :)